frontend/src/app/shared/components/base/base.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, OnDestroy } from '@angular/core';
+import { Observable, Subject, takeUntil } from 'rxjs';
+
+@Component({
+    selector: 'app-base',
+    template: ''
+})
+export class BaseComponent implements OnDestroy {
+    protected destroyed$ = new Subject<void>;
+
+    readonly untilThis = <T>(source: Observable<T>) => source.pipe(takeUntil(this.destroyed$));
```

Comment:
untilThis is not clear. untilDestroyed is more specific

Reply:
Renamed